book/plugins/how-to-guides/maximize-compatibility.md

```diff
@@ -18,13 +18,259 @@ Some guidance on each of these approaches:
+```
+
+With <env-file-branch> being defined as the branch of your plugin's github repository where the environment files are located (we recommend the main branch for this, just to keep things simple).
+
+This method also provides an easy way for users to update their environment any time you create a new release for your plugin on Github.
+Using different branches on your repository as release locations, you can create a new environment file for each release with your plugin's install location reflecting the new target (i.e. release) branch.
+
+As an example, your branch structure could look like the following:
+
+```
+release-2024.5
+release-2024.10
+```
+Which would result in the following environment files (with amplicon as the target distribution):
+
+2024.5
+- Environment file name:
+`2024.5-my-plugin-environment.yml`
+
+- Contents:
+```
+channels:
+- https://packages.qiime2.org/qiime2/2024.5/amplicon/released
+- conda-forge
+- bioconda
+- defaults
+dependencies:
+  - qiime2-amplicon
+  - pip
+  - pip:
+    - my_plugin@git+https://github.com/my-plugin-org/my-plugin.git@release-2024.5
+```
+
+2024.10
+- Environment file name:
+`2024.10-my-plugin-environment.yml`
+
+- Contents:
+```
+channels:
+- https://packages.qiime2.org/qiime2/2024.10/amplicon/released
+- conda-forge
+- bioconda
+- defaults
+dependencies:
+  - qiime2-amplicon
+  - pip
+  - pip:
+    - my_plugin@git+https://github.com/my-plugin-org/my-plugin.git@release-2024.10
+```
+
+Users' initial install command would look like this:
+
+```
+conda env create -n my-plugin -f https://raw.githubusercontent.com/my-plugin-org/my-plugin/main/environment-files/2024.5-my-plugin-environment.yml
+```
+
+With the next release file looking identical aside from the branch name and target release (2024.10), users would then run the following command to update their existing 2024.5 environment:
+
+```
+conda env update -n my-plugin -f https://raw.githubusercontent.com/my-plugin-org/my-plugin/main/environment-files/2024.10-my-plugin-environment.yml
+```
+
+Note that you may want to utilize a different branch location for your environment files than for each of your package releases (i.e. main branch on the repo vs. a specific release branch).
+This will allow for all of your environment files to live in a singular location, for ease of reference.
+
+2. Installing your plugin using the Tiny Distribution and any custom required plugins (not recommended)
+
+If you are working on a unique plugin that is not compatible with one of our existing distributions (amplicon, metagenome) that has a few specific q2 plugin dependencies, you'll utilize a similar approach to install option 1 - just with a more customized environment file.
+As a reminder, while this approach is fairly straightforward to implement, we don't recommend this (if at all possible) because this will be more difficult for us to assist with and help users to troubleshoot.
```

Comment:
```suggestion
As a reminder, while this approach is fairly straightforward to implement, we don't recommend this if the option presented above is possible for your plugin, because this will be more difficult for us to assist with and help users to troubleshoot.
```